tritium-lib/src/main/java/com/palantir/tritium/proxy/Instrumentation.java

```diff
@@ -134,6 +136,27 @@ private Builder(Class<T> interfaceClass, U delegate) {
             return withMetrics(metricRegistry, null);
         }
 
+        /**
+         * Supplies a TaggedMetricRegistry and a name prefix to be used across service invocations.
+         *
+         * Uses a {@link TaggedMetricsServiceInvocationEventHandler} object for handling invocations, so
+         * metric names are chosen based off of the interface name and invoked method.
+         *
+         * @param metricRegistry - TaggedMetricsRegistry used for this application.
+         * @param globalPrefix - Metrics name prefix to be used
+         * @return - InstrumentationBuilder
+         */
+        public Builder<T, U> withTaggedMetrics(TaggedMetricRegistry metricRegistry, String globalPrefix) {
+            checkNotNull(metricRegistry, "metricRegistry");
+            this.handlers.add(new TaggedMetricsServiceInvocationEventHandler(
+                    metricRegistry, globalPrefix));
+            return this;
+        }
+
+        public Builder<T, U> withTaggedMetrics(TaggedMetricRegistry metricRegistry) {
+            return withTaggedMetrics(metricRegistry, null);
```

Comment:
Does this always NPE?

Reply:
yes 👎, see https://github.com/palantir/tritium/pull/153 to fix